rtl/templates/sonata_xbar_main.sv.tpl

```diff
@@ -93,24 +131,14 @@ module sonata_xbar_main
     .tl_xadc_i        (tl_xadc_i),
     .tl_timer_o       (tl_timer_o),
     .tl_timer_i       (tl_timer_i),
-    .tl_spi_board_i   (tl_spi_board_i),
-    .tl_spi_board_o   (tl_spi_board_o),
-    .tl_spi_lcd_i     (tl_spi_lcd_i),
-    .tl_spi_lcd_o     (tl_spi_lcd_o),
-    .tl_spi_ethmac_i  (tl_spi_ethmac_i),
-    .tl_spi_ethmac_o  (tl_spi_ethmac_o),
     % for block in config.blocks:
-    % if not block.name == "gpio":
+    % if block.name not in ["gpio", "i2c", "spi"]:
```

Comment:
I think it may be nicer to add field to a block which states which crossbar it should be in. Instead of hard coding certain blocks in the templates.

Reply:
Fair enough, I've had a go. How does it look to you now?